Common/src/geometry/CGeometry.cpp

```diff
@@ -3637,7 +3638,56 @@ const CCompressedSparsePatternUL& CGeometry::GetEdgeColoring(su2double* efficien
 
     /*--- Color the edges. ---*/
     constexpr bool balanceColors = true;
-    edgeColoring = colorSparsePattern(pattern, edgeColorGroupSize, balanceColors);
+
+    /*--- lambda to account for different numbers of colors ---*/
+    auto getColorSparsePattern = [&](unsigned long edgeColorGroupSize) {
+      if (largeNumberOfColors) {
+        return colorSparsePattern<unsigned char, 255>(pattern, edgeColorGroupSize, balanceColors);
+      } else {
+        return colorSparsePattern(pattern, edgeColorGroupSize, balanceColors);
+      }
+    };
+
+    /*--- if requested, find an efficient coloring with maximum color group size (up to edgeColorGroupSize) ---*/
+    if (maximizeEdgeColorGroupSize) {
+      auto upperEdgeColorGroupSize = edgeColorGroupSize + 1; /* upper bound that is deemed too large */
+      auto nextEdgeColorGroupSize = edgeColorGroupSize;      /* next value that we are going to try */
+      auto lowerEdgeColorGroupSize = 1ul;                    /* lower bound that is known to work */
```

Comment:
const where possible please

Reply:
`upperEdgeColorGroupSize`, `nextEdgeColorGroupSize`, `lowerEdgeColorGroupSize` are all overwritten in the course of the loop that adapts the edge color group size. I found some other places to add `const` though, thanks for the remark :+1: